trestle/core/ssp_io.py

```diff
@@ -213,14 +215,53 @@ def get_control_response(self, control_id: str, level: int) -> str:
         'The System' is the default response, and all other components are treated as sub-headings per response item.
         """
         if not self._profile_path:
-            raise TrestleError('Cannot get control statement, set profile first.')
+            raise TrestleError('Cannot get control response, set profile first.')
 
-        writer = ControlIOWriter()
         control = self._catalog_interface.get_control(control_id)
-        group_id, group_title, _ = self._catalog_interface.get_group_info_by_control(control.id)
-        response_lines = writer.get_response(control)
+        control_impl_req = self._control_implemented_req(control_id)
 
-        tree = MarkdownNode.build_tree_from_markdown(response_lines)
+        md_writer = MDWriter(None)
+        if control_impl_req.statements:
+            for statement in control_impl_req.statements:
+                statement_id = statement.statement_id
+                label = statement_id
+                # look up label for this statement
+                if control.parts:
+                    found_label = self._catalog_interface.get_statement_label_if_exists(control_id, statement_id)
+                    if found_label:
+                        label = found_label
+
+                header = f'Implementation {label}'
+                response_per_component = {}
+                if statement.by_components:
+                    for idx, component in enumerate(statement.by_components):
+                        # look up component title
+                        subheader = component.uuid
+                        response = ''
+                        if self._ssp.system_implementation.components:
+                            for comp in self._ssp.system_implementation.components:
+                                title = comp.title
+                                if title == SSP_MAIN_COMP_NAME and idx == 0:
+                                    # special case dont print first "This system"
+                                    continue
+                                subheader = title
```

Comment:
Something looks strange here. It looks like it missing another comparison operation. (e.g. if component.uuid == comp.uuid) do X.

Reply:
Good catch, fixed